pkg/controller/machineset.go

```diff
@@ -677,6 +694,75 @@ func (c *controller) terminateMachines(inactiveMachines []*v1alpha1.Machine, mac
 	return nil
 }
 
+func (c *controller) terminateStaleMachines(inactiveMachines []*v1alpha1.Machine, deletionWindow int, machineSet *v1alpha1.MachineSet) error {
+	var (
+		wg    sync.WaitGroup
+		errCh = make(chan error, deletionWindow)
+	)
+	defer close(errCh)
+
+	wg.Add(deletionWindow)
+	for i := 0; i < deletionWindow; i++ {
+		go c.prepareMachineForDeletion(inactiveMachines[i], machineSet, &wg, errCh)
```

Comment:
@prashanth26 Could you please check, from your previous fix, if there is a scope of improvement. 
I fear, if only few of the threads fail here, rest are mid way, and later in new reconcilliation new set of stale-machines are chosen for deletion, we may break the contract for ratio.

Reply:
I think failing here should be fine. Considering that we are using waitgroups anyways for all the threads to complete right? WDYT?